auth/credentials/impersonate/impersonate.go

```diff
@@ -62,55 +65,49 @@ func NewCredentials(opts *CredentialsOptions) (*auth.Credentials, error) {
 
 	var client *http.Client
 	var creds *auth.Credentials
-	if opts.Client == nil && opts.Credentials == nil {
+	if opts.Client == nil {
 		var err error
-		creds, err = credentials.DetectDefault(&credentials.DetectOptions{
-			Scopes:           []string{defaultScope},
-			UseSelfSignedJWT: true,
-		})
-		if err != nil {
-			return nil, err
+		if opts.Credentials == nil {
+			creds, err = credentials.DetectDefault(&credentials.DetectOptions{
+				Scopes:           []string{defaultScope},
+				UseSelfSignedJWT: true,
```

Comment:
Pass through UD here too.

Reply:
I thought about doing this, but I don't think we should, since the value provided would override a `universe_domain` value in the detected credentials (see `resolveUniverseDomain` in `filetypes.go`).

The problem is that the UD provided here is the *client* UD, which must be compared against the *credentials* UD. Users should not normally need to provide a *credentials* UD in code (override) to `DetectDefault`. But if they do, I think they should then call `DetectDefault` externally with the provided UD, then pass the returned credentials to the impersonate call.

Reply:
TL;DR: The *client* UD available here in `impersonate.CredentialsOptions` should never be used as a *credentials* UD.